File: cmd/2025/day_1/main.go
package main

import (
	"fmt"
	"strings"

	"github.com/Palaszontko/advent-of-code/cmd/utils"
)

func main() {
	fmt.Println("Advent of Code 2025!")
	Part1()
	// Part2()
}

func Part1() {
	fmt.Println("Part 1")

	input := utils.ReadFile("cmd/2025/day_1/input.txt")

	start := 50
	amount := 0

	for _, code := range strings.Split(input, "\n") {
		letter, value := code[0], utils.MustAtoi(code[1:])

		value %= 100

		if letter == 'L' {
			if start-value < 0 {
				start = 100 - (value - start)
			} else {
				start = (start - value) % 100
			}
		} else {
			start = (start + value) % 100
		}

		if start == 0 {
			amount += 1
		}

	}

	fmt.Println(amount)
}

func Part2() {
	fmt.Println("Part 2")
}

File: rust/2025/day_1/src/main.rs
use std::time::Instant;

use aoc_2025_day_1::{part1, part2};

fn main() {
    println!("Advent of Code 2025 - day 1");
    let input = include_str!("../input.txt").trim_end_matches('\n');

    let t = Instant::now();
    let p1 = part1(input);
    println!("Part 1: {p1}  ({:?})", t.elapsed());

    let t = Instant::now();
    let p2 = part2(input);
    println!("Part 2: {p2}  ({:?})", t.elapsed());
}
